### PR TITLE
Refactor units detection used as weight/dimensions

### PR DESCRIPTION
**Pull request by joiggama**
- Default to metric system unless specified
- Allow Quantified instances as initialization arguments e.g. Mass,
  Length
- Refactor detection of system and helper mechanisms

### CODE REVIEW

File: lib/omniship/package.rb [hunk 1]
```diff
@@ -117,7 +106,46 @@ def attribute_from_metric_or_imperial(obj, klass, metric_unit, imperial_unit)
         return klass.new(obj, (@unit_system == :imperial ? imperial_unit : metric_unit))
       end
     end
-    
+
+    def determine_dimensions(values)
+      values.reject!(&:nil?)
```
Comment (mumoc) [inline lib/omniship/package.rb]: not sure, but you could use `compact` if you're removing nil values

Comment (mumoc) [inline lib/omniship/package.rb]: or you could use `values.reject!(&:blank?)` which will remove nil, "", "\n", " ", "\n\r" which will help you on removing the dimention = 0 if dimension blank? as in the end you're filling the remaining values

Comment (joiggama) [reply lib/omniship/package.rb]: I like the latter!, thanks.

**Comment by mumoc** [comment]
LGTM

**Pull request merged**